### ISSUE TITLE
countLeadingZeros for Fixed

### ISSUE THREAD

**Issue by tomjaguarpaw** [clash-lang/clash-compiler#160]
`countLeadingZeros` does not currently work for `Fixed`.  My workaround is to convert to `Int32` or `Int64` and use `countLeadingZeros` on that, which does work.

**Comment by christiaanb** [comment]
But only for `Fixed` numbers requiring fewer than 65 bits, correct?

**Comment by tomjaguarpaw** [comment]
Yes.  If I had a bigger `Fixed` I guess I would split it into several `Int64`s.

**Comment by martijnbastiaan** [comment]
@leonschoorl This is fixed in https://github.com/clash-lang/clash-compiler/pull/501 right?

**Comment by leonschoorl** [comment]
Well, `countLeadingZeros` on Fixed works in general.

Only constant folding doesn't work
